extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheMethodValidator.java

```diff
@@ -0,0 +1,122 @@
+package io.quarkus.cache.deployment;
+
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_INVALIDATE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_INVALIDATE_ALL;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_LOAD;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_STORE;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.NON_BLOCKING_RETURN_TYPES;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.Type;
+
+import io.quarkus.arc.processor.AnnotationStore;
+import io.quarkus.arc.processor.BeanInfo;
+import io.quarkus.cache.deployment.exception.MultipleCacheAnnotationsException;
+import io.quarkus.cache.deployment.exception.NoParameterException;
+import io.quarkus.cache.deployment.exception.NonBlockingReturnTypeException;
+import io.quarkus.cache.deployment.exception.VoidReturnTypeException;
+
+public class CacheMethodValidator {
+
+    public static void validateAnnotations(AnnotationStore annotationStore, BeanInfo bean, MethodInfo method,
+            List<Throwable> throwables) {
+
+        int methodAnnotationsCount = 0;
+        if (validateCacheLoadAnnotation(annotationStore, bean, method, throwables)) {
+            methodAnnotationsCount++;
+        }
+        if (validateCacheStoreAnnotation(annotationStore, bean, method, throwables)) {
+            methodAnnotationsCount++;
+        }
+        if (validateCacheInvalidateAnnotation(annotationStore, bean, method, throwables)) {
+            methodAnnotationsCount++;
```

Comment:
It sounds like something possible, but it could also be achieved by calling a `@CacheLoad`-annotated method and then a `@CacheInvalidate[All]`-annotated method.

Reply:
Speaking of limiting the number of cache annotations per method, I just realized this limitation isn't mentioned in the current Javadoc. I'll update the Javadoc if the limitation check is confirmed.

Reply:
Agree with @gwenneg that the user call a method and then the other. @gsmet In what you'd case you'd want quarkus to do that for you?

Reply:
I updated the javadoc to add a sentence about the one cache annotation per method limitation. It can still be removed if we agree on removing this check. I just wanted the current javadoc to reflect the exact behavior of the code.